concourse/steps/scan_container_images.py

```diff
@@ -183,12 +184,24 @@ def comp_filter(component):
         raise NotImplementedError()
 
 
-def virus_scan_images(image_references: typing.Iterable[str]):
+def virus_scan_images(image_references: typing.Iterable[str], clamav_config_name: str):
+    clamav_client = ccc.clamav.client_from_config_name(clamav_config_name)
     for image_reference in image_references:
-        status, signature = clamav.util.scan_container_image(image_reference=image_reference)
-        if clamav.util.result_ok(status=status, signature=signature):
+        scan_result = clamav_client.scan_container_image(image_reference=image_reference)
+        if clamav_client.result_ok(scan_result):
             continue
-        yield (image_reference, f'{status}: {signature}')
+
+        if isinstance(scan_result, ClamAVScanResult):
```

Comment:
hmm.. I think, it is not ideal style to use runtime type inspection for this kind of flow control.

To me, it looks it would be more appropriate to raise `ClamAVError` (make it an exception). You could still handle the special case (HTTP-422) within the exception handling block, otherwise just re-raise